code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -171,18 +224,31 @@ LINEN BINS
 	icon_state = "sheetsyndie"
 	item_color = "syndie"
 
+/obj/item/weapon/bedsheet/syndie/Initialize()
+	.=..()
+	dreamMessage = list("Get dat fukkin disk", "Telecrystals", "Die Gloriously", "Powersink", "Emag", "E-sword")
+
 /obj/item/weapon/bedsheet/cult
 	name = "cultist's bedsheet"
 	desc = "You might dream of Nar'Sie if you sleep with this. It seems rather tattered and glows of an eldritch presence."
 	icon_state = "sheetcult"
 	item_color = "cult"
 
+/obj/item/weapon/bedsheet/cult/Initialize()
+	.=..()
+	dreamMessage = list("Nar-sie", "Runes", "Arcane Tome", "</i><span class='narsie'>NAR-SIE HAS RISEN</span><i>")
+	dreamSound = 'sound/hallucinations/im_here1.ogg'
+
 /obj/item/weapon/bedsheet/wiz
 	name = "wizard's bedsheet"
 	desc = "A special fabric enchanted with magic so you can have an enchanted night. It even glows!"
 	icon_state = "sheetwiz"
 	item_color = "wiz"
 
+/obj/item/weapon/bedsheet/wiz/Initialize()
+	.=..()
+	dreamMessage = list("EI NATH!", "Hocus Pocus", "Casting magic missile at the darkness", "Beards", "Staff", "Robe and wizard hat", "Sandals", "UNLIMITED POWER!")

```

Comment:
these don't need to be in Initialize either, they can just be on the object.
same for the sound, which could probably also be a list of sounds